File: app/src/main/java/com/pr0gramm/app/ui/views/viewer/video/VideoPlayer.java
package com.pr0gramm.app.ui.views.viewer.video;

import android.net.Uri;
import android.support.annotation.Nullable;

/**
 */
public interface VideoPlayer {
    void setVideoCallbacks(@Nullable VideoPlayer.Callbacks callbacks);

    void open(Uri uri);

    void start();

    float progress();

    float buffered();

    void pause();

    void rewind();

    boolean isMuted();

    void setMuted(boolean muted);

    interface Callbacks {
        void onVideoBufferingStarts();

        void onVideoBufferingEnds();

        void onVideoRenderingStarts();

        void onVideoSizeChanged(int width, int height);

        void onVideoError(String message, ErrorKind kind);

        void onDroppedFrames(int count);
    }

    enum ErrorKind {
        UNKNOWN,
        NETWORK
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/views/viewer/video/VideoPlayer.kt
package com.pr0gramm.app.ui.views.viewer.video

import android.net.Uri

/**
 */
interface VideoPlayer {

    val progress: Float

    val buffered: Float

    val duration: Int

    val currentPosition: Int

    var videoCallbacks: VideoPlayer.Callbacks?

    var muted: Boolean

    fun open(uri: Uri)

    fun start()

    fun pause()

    fun rewind()

    fun seekTo(position: Int)

    interface Callbacks {
        fun onVideoBufferingStarts()

        fun onVideoBufferingEnds()

        fun onVideoRenderingStarts()

        fun onVideoSizeChanged(width: Int, height: Int)

        fun onVideoError(message: String, kind: ErrorKind)
    }

    enum class ErrorKind {
        UNKNOWN,
        NETWORK
    }
}
